packages/web/contexts/GameContext.tsx

```diff
@@ -0,0 +1,328 @@
+  fetchGameData: async () => {},
+  visitedElements: () => '0',
+  mintChiev: async () => '',
+  connect: async () => undefined,
+  disconnect: async () => undefined,
+  loading: true,
+  txLoading: false,
+  account: '',
+  network: '0x013881',
+  connected: false,
+  connecting: false,
+});
+
+export const GameContextProvider: React.FC = ({ children }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [txLoading, setTxLoading] = useState(false);
+  const [gameDataState, setGameDataState] = useState<GamePropertiesType>();
+  const {
+    address,
+    provider,
+    connect,
+    connecting,
+    connected,
+    chainId,
+    disconnect,
+  } = useWeb3();
+  const [account, setAccount] = useState<any>(address ?? '');
+  const toast = useToast();
+
+  /** Function to async fetch `CONFIG.onboardingGameDataURL` as json and return the data
+   * TODO: this needs the func from the main Game.tsx file to be moved here to replace
+   * this function
+   */
+  const fetchGameData = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      console.log('Fetchng GameData...');
+
+      // const response = await fetch(gameJson);
+      // const data = (await response.json()) as GameProperties;
+      const data = gameJson as GameProperties;
+      console.log('fetchGameData', data);
+
+      if (data) {
+        setGameDataState(data);
+      }
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error);
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (address) {
+      setAccount(address);
+    }
+  }, [address]);
+
+  const game = useMemo((): GameProperties => {
+    try {
+      if (!isLoading && gameDataState) {
+        const {
+          assets,
+          name,
+          startingElement,
+          elements,
+          connections,
+          components,
+        } = gameDataState;
+        return {
+          name,
+          assets,
+          startingElement,
+          elements,
+          connections,
+          components,
+        } as GameProperties;
+      }
+      return {
+        name: '',
+        assets: {},
+        startingElement: '',
+        elements: {},
+        connections: {},
+        components: {},
+      } as GameProperties;
+    } catch (error) {
+      console.log('fetchGameData error: ', { error });
+      return {
+        name: '',
+        startingElement: '',
+        assets: {},
+        elements: {},
+        connections: {},
+        components: {},
+      } as GameProperties;
+    }
+  }, [gameDataState, isLoading]);
+
+  /** Function to get, set or clear game state
+   *
+   * @param currentPlace - the current place of the player
+   * @param reset - if true, clear the game state
+   * @returns the game state or undefined if reset is true or the game state is not set
+   */
+  const gameState = (
+    currentPlace?: string,
+    reset?: boolean,
+  ): IGameState['state'] => {
+    if (currentPlace !== undefined) {
```

Comment:
It is a string, so this is likely safe.

```suggestion
    if (currentPlace) {
```